### DISCUSSION TITLE
is there a rate limit?

### DISCUSSION THREAD

**Discussion by dragitz** [sbidy/wiz_light#112] [Q&A]
Hi, I'm making a program that modifies the brightness of the bulb (e27 works well).
The input is a byte value that changes between 0 and 255 so it's perfect. (using controller stick as input)

My program sends this command via cmd every 200ms (X = 0-255)
wizlight on --ip 192.168.0.18 --k 2200 --brightness X

After a while the light gets frozen but on the phone app I can freely change the brightness. is there a rate limit?

also I just send the coomand without waiting for a reply, maybe that's the problem, but by doing that it's a slow process

**Comment by sbidy** [comment]
yes - you cant trigger the command so often. You should update the state only every second. This because of the connection via UDP.
The app uses MQTT via TCP over the cloud connection.

Overall you should check the state every time you change it. Overall the bulbs are not designed to process multiple state changes in seconds (because of power spikes).

**Reply by dragitz** [reply]
Thank you for the reply! I've set the cooldown to 1second and it works like a charm

**Reply by vodovozovge** [reply]
> The app uses MQTT via TCP over the cloud connection.

it actually should try UDP first if the bulb replied to the registration packet when the app is launched.
200 ms should not be an issue for the bulb itself (we rate limit to 200 ms I think in the WiZ App for the controls like brightness or custom color). Interesting observation about the bulb being frozen after a while, let me try to reproduce it
